src/Controller/FightersController.php

```diff
@@ -14,11 +14,37 @@ public function index(): string
         return $this->twig->render('Fighters/fighterslist.html.twig', ['fighters' => $fighters]);
     }
 
-    public function pick(int $id): string
+    public function pick(): string
     {
+        // set fighters
+
         $fighterManager = new FighterManager();
-        $fighter = $fighterManager->selectOneById($id);
+        $fighters = $fighterManager->selectAll('id');
+
+        if (isset($_SESSION['player1']) && isset($_SESSION['player2'])) {
+            header('Location: /fight/attack');
+        }
+
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            if (!isset($_SESSION['player1'])) {
+                $_SESSION['player1'] = $fighterManager->selectOneById($_POST['selected_fighter']);
+                header('Location: /fight/pick');
+            } elseif (!isset($_SESSION['player2'])) {
+                $_SESSION['player2'] = $fighterManager->selectOneById($_POST['selected_fighter']);
+                header('Location: /fight/attack');
+            }
+        }
+
+        // initiate fighters
+
+        $player1 = $_SESSION['player1'] ?? null;
+        $player2 = $_SESSION['player2'] ?? null;
+
+        $_SESSION['player1'] = $player1;
+        $_SESSION['player2'] = $player2;
+        $currentAttacker = $player1;
+        $_SESSION['currentAttacker'] = $currentAttacker;
```

Comment:
I'm pretty sure you can remove this part: $_SESSION['player1'] and $_SESSION['player2'] are handled in the conditions, and $_SESSION['currentAttacker'] could (should?) be handled in the conditions as well